File: WEEKLYCODE#4/designer_pdf_viewer/Java/designer_pdf_viewer.java
import java.util.Scanner;
import java.lang.Math;

public class Solution {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int[] heights  = new int[26];
    
        //taking inputs
        for (int i = 0; i < 26; i++) {
            heights[i] = input.nextInt();
        }
        String word = input.next();
        input.close();
        
        int rect_height = 0;
        
        //calculating max height
        for (int j = 0; j < word.length(); j++) {
            rect_height = Math.max(
                rect_height,
                heights[(int)word.charAt(j) - (int)'a']
            );
        }
        
        //printing answer
        int ans = word.length() * rect_height;
        System.out.println(ans);
    }
}

File: WEEKLYCODE#4/designer_pdf_viewer/C/designer_pdf_viewer.c
#include<stdio.h>
#include<string.h>
int main(){
    int ar[26],i;
    //taking the array of heights as input
    for(i=0;i<26;i++)
        scanf("%d",&ar[i]);
    char str[15];
    //taking string input
    scanf("%s",str);
    //finding max height in the word
    int max=ar[str[0]-97];
    for(i=1;i<strlen(str);i++)
        if(max<ar[str[i]-97])
            max=ar[str[i]-97];
    //prnting answer
    printf("%d",max*strlen(str));
    return 0;
}
